src/main/java/org/apache/commons/pool3/impl/BaseGenericObjectPool.java

```diff
@@ -1617,7 +1624,8 @@ public final void setTestWhileIdle(final boolean testWhileIdle) {
      * @param delay duration before start and between eviction runs.
      */
     final void startEvictor(final Duration delay) {
-        synchronized (evictionLock) {
+        try {
+            evictionLock.lock();
```

Comment:
locking should be done before try block to avoid unlocking not locked lock when locking method throws an exception

Reply:
the very same goes for other places in code

Reply:
done

Reply:
thank you 👍🏼